velox/exec/WindowPartition.h

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "velox/exec/RowContainer.h"
+#include "velox/vector/BaseVector.h"
+
+namespace facebook::velox::exec {
+class WindowPartition {
+ public:
+  WindowPartition(
+      const std::vector<exec::RowColumn>& columns,
+      const std::vector<TypePtr>& types);
+
+  /// Extracts into the result VectorPtr values from column at index 'idx'
+  /// for numRows starting at rowOffset in the partition.
+  /// This API is useful for window functions that examine column values
+  /// for their logic. Nulls at corresponding positions are copied.
+  void extractColumn(
```

Comment:
Is this method used? If not, perhaps, remove it from this PR and add it in a follow-up PR alongside usage and test coverage.